src/storage/src/hummock/sstable/block_iterator.rs

```diff
@@ -41,13 +42,26 @@ pub struct BlockIterator {
     last_key_len_type: LenType,
     last_value_len_type: LenType,
 
-    /// NOTE: `hitmap` is supposed be updated every time when `value_range` is updated.
-    hitmap: LocalHitmap<HITMAP_ELEMS>,
+    /// NOTE:
+    ///
+    /// - `hitmap` is supposed to be updated each time accessing the block data in a new position.
+    /// - `hitmap` is supposed to be dropped before `block`
+    hitmap: ManuallyDrop<LocalHitmap<HITMAP_ELEMS>>,
```

Comment:
The current way of report hitmap leverages the `drop` of local hitmap, but the correctness of the `drop` of local hitmap should depend on the correct implementation of the `drop` of `BlockIterator`. If so, I think we'd better decouple hitmap and local hitmap, which means we won't associate a hitmap the local hitmap, and the local hitmap won't implement its drop. We only manually apply the local hitmap to the block hitmap in the `drop` of `BlockIterator`.

Reply:
Or I have another idea.